File: Q9/Q9.go
package main

import (
    "fmt"
)

func main() {
    var num int
    isPrime := true

    fmt.Print("Enter a positive integer: ")
    fmt.Scan(&num)

    if num <= 1 {
        isPrime = false
    } else {
        for i := 2; i <= num/2; i++ {
            if num%i == 0 {
                isPrime = false
                break
            }
        }
    }

    if isPrime {
        fmt.Printf("%d is a prime number.\n", num)
    } else {
        fmt.Printf("%d is not a prime number.\n", num)
    }
}

File: Q9/Q9.c
#include <stdio.h>

int main() {
    int num, i, isPrime = 1;

    printf("Enter a positive integer: ");
    scanf("%d", &num);

    if (num <= 1) {
        isPrime = 0;
    } else {
        for (i = 2; i <= num / 2; i++) {
            if (num % i == 0) {
                isPrime = 0;
                break;
            }
        }
    }

    if (isPrime)
        printf("%d is a prime number.\n", num);
    else
        printf("%d is not a prime number.\n", num);

    return 0;
}
